source/common/stats/stat_merger.cc

```diff
@@ -15,15 +13,34 @@ void StatMerger::mergeCounters(const Protobuf::Map<std::string, uint64_t>& count
 
 void StatMerger::mergeGauges(const Protobuf::Map<std::string, uint64_t>& gauges) {
   for (const auto& gauge : gauges) {
+    // Merging gauges via RPC from the parent has 3 case; case 2 and 4b are the
+    // most common.
+    //
+    // 1. Parent process thinks gauge is NeverImport: no data sent, and we
+    //    do not run this loop for such a gauge. Only if the parent process
+    //    thinks the gauge is Accumulate do we consider cases 2-4.
+    // 2. Child thinks gauge is Accumulate : data is combined in
+    //    gauge_ref.add() below.
+    // 3. Child thinks gauge is NeverImport: we skip this loop entry via
```

Comment:
Can you clarify that this case only happens if a gauge has been changed to NeverImport by the child? Is that right?

Reply:
yes, added comment.